### PR TITLE
Authn chain elements

### PR DESCRIPTION
**Pull request by lobkovilya**
Depends on #152 

## Description 
Add 2 chain elements:
- `AuthnClient` adds token to the current `PathSegment` (pointed by `Index`)
- `AuthnServer` verifies token  of the current `PathSegment`

### CODE REVIEW

**Comment by denis-tingajkin** [comment]
@edwarnicke I think this PR can be useful for **Advanced OPA Policies**. Can you please take a look?

File: pkg/networkservice/common/authn/server.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/security"
+)
+
+type authnServer struct {
+	p security.Provider
```
Comment (edwarnicke) [inline pkg/networkservice/common/authn/server.go:37]: I really don't want to be promulgating additional APIs around... much of the stuff in the security package isn't needed now that we have grpcoptions.

File: pkg/networkservice/common/authn/server.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+func (a *authnServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	x509crt, err := peerCertificateFromContext(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, "unable to get peer's TLS cert from context: %v", err)
+	}
+
+	token := request.GetConnection().GetPath().GetPathSegments()[request.GetConnection().GetPath().GetIndex()].GetToken()
+	if err := security.VerifyToken(token, x509crt); err != nil {
```
Comment (edwarnicke) [inline pkg/networkservice/common/authn/server.go:55]: Can't we do the token verification in OPA?

Comment (denis-tingajkin) [reply pkg/networkservice/common/authn/server.go:55]: I think it is a good idea, will do :)

**Pull request closed without merging**